internal/epoch/epoch_manager.go

```diff
@@ -458,10 +460,14 @@ func (e *Manager) refreshLocked(ctx context.Context) error {
 	return e.maybeCompactAndCleanup(ctx, *p)
 }
 
-func (e *Manager) maybeCompactAndCleanup(ctx context.Context, p Parameters) error {
+func (e *Manager) allowWrites() bool {
 	// Disable compaction and cleanup operations when running in read-only mode
 	// since they'll just fail when they try to mutate the underlying storage.
-	if e.st.IsReadOnly() {
+	return e.allowCleanupWritesOnIndexLoad && !e.st.IsReadOnly()
```

Comment:
This is effectively disabling index maintenance on load, since the default value for `allowCleanupWritesOnIndexLoad` is `false`.